proposals/0051-serialized-manifest.md

```diff
@@ -0,0 +1,169 @@
+- Information that can be used directly by our end-users. Integration developers have access to `AstroConfig`.
+- Information that isn't present inside `AstroConfig`. As mentioned previously, the `SSRManifest` contains useful information that are often used by on-demand pages and adapters.
+  An example is `SSRManifest.i18n.domainLookupTable`, which is used by adapters to understand the mapping of a domain to a locale.
+
+These are guidelines, so exceptions are possible, as long as they have a good reason to be exposed. 
+
+## Proposed APIs
+
+Proposed name of the module: `astro:config`
+
+Proposed submodules
+
+### `astro:config/client`
+
+Exported bindings:
+
+  - `i18n`
+  - `trailingSlash`
+  - `base`
+  - `build.format`
+  - `site`
+  - `redirects`[^1]
+
+### `astro:config/server`
+
+Exported bindings:
+  - `srcDirc`
+  - `cacheDir`
+  - `outDir`
+  - `build.client`
+  - `build.server`
+  - `build.serverEntry`
+  - `build.assetsPrefix`
+  - `publicDir`
+  - `root`
+  - `i18n`
+  - `trailingSlash`
+  - `base`
+  - `build.format`
+  - `site`
+  - `redirects`[^1]
+
+
+To note that `/server`/`/client` can be used anywhere in the code, even in client scripts.
+
+[^1]: To evaluate, probably not needed
+
+## Deprecations
+
+We will deprecate existing means to read this information. The following will be deprecated:
+- `import.meta.env.BASE_URL`
+- `import.meta.env.SITE`
+- `import.meta.env.ASSETS_PREFIX`
+- `Astro.site`/`ctx.site`
+
+# Testing Strategy
+
+- The virtual modules will be "usable" via experimental flag. Attempting to use these virtual modules without the experimental flag turned on will result into a hard error.
+- After a gracing period where we stabilise the APIs, the experimental flag will be removed.
+
+
+# Drawbacks
+
+Exposing these kinds of information requires "drawing a line" for when something can be exposed to our end-users, instead of telling them to use an integration.
+
+I believe it will be more difficult to come up with important and firm guidelines, but once we have them, maintainers will know when something is worth exposing to the users.
+
+# Alternatives
+
+Many of the information available to the user are related to routing e.g. `trailingSlash`, `build.format`, `i18n`, `base`, etc. 
+so I also evaluated exposing *utility functions* to create links, however we don't know what are the use cases, and even if we did, there are many other information at play,
+such as the pipeline (dev VS build VS on-demand pages), prerendering, server islands, and more. 
+
+I feel that Astro should provide the primitives to integrations, so they can create utility functions based on their use cases.
+
+# Adoption strategy
+
+- The experimental flag `experimental.serializeConfig` will be shipped in a minor.
+- Once the RFC is stable and approved, a new minor will remove the flag.
```

Comment:
We can probably add a line about adding the deprecations with the minor

Reply:
Also about removing the deprecated APIs in a future major

Reply:
I added the deprecations as non-goals since we have different visions for this new feature. Deprecations can added later